components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/endpoint/registry/util/EndpointRegistryUtil.java

```diff
@@ -32,4 +46,24 @@ public static void raiseResourceAlreadyExistsException(String msg) throws
         throw new EndpointRegistryResourceAlreadyExistsException(msg);
     }
 
+    public static boolean isValidDefinition(URL definitionURL, String definitionType)
+            throws DefinitionValidationException {
+        DefinitionValidator definitionValidator = definitionValidatorMap.get(definitionType);
+        if (definitionValidator != null) {
+            return definitionValidator.validate(definitionURL);
+        }
+        throw new DefinitionValidationException("No Definition Validator found for the given definition type: '"
+                + definitionType + "'");
+    }
+
+    public static boolean isValidDefinition(byte[] definitionContent, String definitionType)
```

Comment:
javadocs here

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/8618/commits/169ace7202aa557c7b6710f5bf9e481af4ff99d4